core/src/main/scala/org/apache/spark/storage/DiskBlockObjectWriter.scala

```diff
@@ -272,6 +274,25 @@ private[spark] class DiskBlockObjectWriter(
     file
   }
 
+  /**
+   * Reverts write metrics that haven't been committed yet and delete the file held
+   * by current `DiskBlockObjectWriter`. Callers should invoke this function when there
+   * are runtime exceptions in file writing process and the file is no longer needed.
+   */
+  def closeAndDelete(): Unit = {
```

Comment:
We should test this in `DiskBlockObjectWriterSuite` and I think the metrics was wrong and needs to be fixed.

Reply:
`metrics` keeps the original logic now, but I also think it is wrong. I will give feedback later because there are other work to be done today.